File: Flora backend/Flora.Services/Database/DonationCampaign.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Flora.Services.Database
{
    public class DonationCampaign
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime EndDate { get; set; }
        public string? ImageUrl { get; set; }

        public virtual ICollection<Donation> Donations { get; set; } = new List<Donation>();
    }
}

File: Flora frontend/flora_mobile_app/lib/models/donation_campaign.dart
class DonationCampaign {
  final int id;
  final String title;
  final String description;
  final DateTime endDate;
  final String? imageUrl;
  final double totalAmount;

  DonationCampaign({
    required this.id,
    required this.title,
    required this.description,
    required this.endDate,
    this.imageUrl,
    required this.totalAmount,
  });

  factory DonationCampaign.fromJson(Map<String, dynamic> json) {
    return DonationCampaign(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      endDate: DateTime.parse(json['endDate']),
      imageUrl: json['imageUrl'],
      totalAmount: json['totalAmount']?.toDouble() ?? 0.0,
    );
  }
}
